src/mono/System.Private.CoreLib/src/System/GC.Mono.cs

```diff
@@ -116,8 +116,7 @@ public static void KeepAlive(object? obj)
         public static int GetGeneration(WeakReference wo)
         {
             object? obj = wo.Target;
-            if (obj == null)
-                throw new ArgumentException(null, nameof(wo));
+            ArgumentNullException.ThrowIfNull(obj, nameof(wo));
```

Comment:
Is use of `ArgumentNullException` the most appropriate exception, given that the parameter `wo` is not null?

Reply:
It's the same exception coreclr already throws for it here:
https://github.com/dotnet/runtime/blob/4f63db1b9fbee35e3749ab7deb949cd8de63e848/src/coreclr/vm/comutilnative.cpp#L826-L827